### ISSUE TITLE
Git.detached broken in 2017.7.0

### ISSUE THREAD

**Issue by zebooka** [saltstack/salt#42381]
### Description of Issue/Question
States with 'rev' argument fail. There are some mentions of `ref` argument in sources, although there is no such arg anymore.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

### Steps to Reproduce Issue

```
ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1837, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1794, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 2216, in detached
    if len(ref) <= 40 \
TypeError: object of type 'NoneType' has no len()
```
```
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1837, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1794, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 2422, in detached
    if 'refs/remotes/'+remote+'/'+ref in all_remote_refs:
TypeError: cannot concatenate 'str' and 'NoneType' objects
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2017.7.0
 
Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.8 
         locale: UTF-8
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.8 
```

**Comment by jalandis** [comment]
I have been able to get past this by providing both a rev and ref argument with the same values.

```
  - rev: v3.1.0
  - ref: v3.1.0
```

**Comment by rallytime** [comment]
Thanks @zebooka - It looks like we have a reference to the old `ref` that should be `rev`. I'll look into this.

**Comment by rallytime** [comment]
@zebooka and @jalandis This should be fixed in #42399. Please give that a try and let me know how it goes.

**Comment by vijayinvites** [comment]
@jalandis How do you pass the -ref and -rev arguments?
